src/core/hard_forks.zig

```diff
@@ -0,0 +1,85 @@
+const std = @import("std");
+const ArrayList = std.ArrayList;
+const allocator = std.mem.Allocator;
+const Slot = @import("slot.zig").Slot;
+pub const HardFork = struct { Slot, usize };
+pub const HardForks = struct {
+    hard_forks: ArrayList(HardFork),
+
+    const Self = @This();
+    pub fn default(alloc: allocator) Self {
+        return .{ .hard_forks = ArrayList(HardFork).init(alloc) };
+    }
+    pub fn register(self: *Self, new_slot: Slot) !void {
```

Comment:
kinda a bit out of scope for the PR but could you change the `Slot` struct to just be `pub const Slot = u64` - should clean up the code a lot by removing `.value` checks

Reply:
@0xNineteen this is major change, it would affect other parts of the code where slot is used, probably not a good idea to do it in this PR itself

Reply:
@anoushk1234  I think its only used in `crds.zig` rn as default values? - shouldnt be too bad

Reply:
Just to clarify you want to remove Slot as a struct altogether since you wrote `pub const Slot = u64`
```zig
pub const Slot = struct {
    value: u64,

    const Self = @This();

    pub fn init(slot: u64) Self {
        return Self{
            .value = slot,
        };
    }

    /// returns slot with 0 as value
    pub fn default() Self {
        return Self{ .value = 0 };
    }
};
```

Reply:
@anoushk1234 yeah exactly

Reply:
@0xNineteen Zig doesnt support this yet, using an enum is an option though although it's not the clean

Reply:
@0xNineteen done